cli/init.go

```diff
@@ -0,0 +1,441 @@
+package cli
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const (
+	githubTarballURL = "https://github.com/%s/%s/tarball/%s"
+	githubAPIURL     = "https://api.github.com/repos/%s/%s/branches/%s"
+	owner            = "dispatchrun"
+	repo             = "dispatch-examples"
+	branch           = "main"
+	dispatchUserDir  = "dispatch"
+)
```

Comment:
A personal preference, so feel free to keep it as-is if you think it's clearer: I would inline these string literals instead of using constants, especially if they're used in only one place. This makes searching easier since there's no indirection (e.g., using `grep` with the URL, it pinpoints the line of code right away).